pkg/http/api.go

```diff
@@ -1707,7 +1707,7 @@ func (a *api) onDirectMessage(reqCtx *fasthttp.RequestCtx) {
 			} else {
 				resStatus.Code = statusCode
 			}
-		} else if resStatus.Code < 200 || resStatus.Code > 299 {
+		} else if resStatus.Code < 200 || resStatus.Code > 399 {
```

Comment:
This change is not closely related to this issue. Since 3xx can not be recovered by the retry defined by the resiliency, I add this change.

Reply:
Makes sense - with this change redirects are valid responses that should not be retried.